File: src/data/SingleAppointment.java
package data;

import java.util.Date;

public class SingleAppointment extends Appointment {

  private Date date;

  public SingleAppointment(Therapist therapist, Patient patient, String time, Date date) {
    super(therapist, patient, time);
    this.date = date;
  }

  public Date getDate() {
    return date;
  }

  public void setDate(Date date) {
    this.date = date;
  }

}

File: frontend/src/class/SingleAppointment.ts
import Appointment from './Appointment';
import { Time } from './Enums';
import Patient from './Patient';
import Therapist from './Therapist';

export default class SingleAppointment extends Appointment {
  date : Date;

  constructor(
    therapist : Therapist,
    patient: Patient[],
    time: Time,
    date: Date,
  ) {
    super(therapist, patient, time);
    this.date = date;
  }
}
